src/Content/ContentStorageHandler.php

```diff
@@ -6,6 +6,7 @@
 use Kirby\Cms\Language;
 use Kirby\Cms\ModelWithContent;
 use Kirby\Cms\Page;
+use Kirby\Exception\NotFoundException;
 
 /**
  * Abstract for content storage handlers;
```

Comment:
`General thought/idea`: The `ContentStorageHandler` abstract keeps getting more complex, which could make it harder to implement custom handlers. Once we have the methods cleaned up and moved to the right places, it might make sense to add an interface again. We would then have `abstract class ContentStorageHandler implements ContentStorageHandlerInterface` and type-hint the interface where we currently type-hint the abstract. Plugin devs who want to use our helpers in the abstract can extend the abstract and those who want to start fresh can implement the interface directly.

Reply:
Sounds good to me. I still think that we can clean up the abstract class quite a bit later. But I like the idea of the interface as a baseline.

Reply:
I've added a todo to Notion.